### PR TITLE
Update serde features to new schema

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl From<RuntimeError> for InterpreterError {
 }
 
 #[derive(Debug, Error)]
-#[cfg_attr(feature = "serde-types-minimal", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```
Comment (kayagokalp) [inline src/error.rs:103]: I think `cargo test --features serde` fails with `the trait "Serialize" is not implemented for "std::io::Error"` and `the trait "Deserialize<'_>" is not implemented for "std::io::Error"`

Comment (kayagokalp) [reply src/error.rs:103]: And as far as I can see CI does not test for feature `serde`

**Comment by kayagokalp** [comment]
@vlopes11 just a heads up I merged the #117 into here, after [this](https://github.com/FuelLabs/fuel-vm/pull/117#issuecomment-1127214031) comment.

**Comment by vlopes11** [comment]
Removed  serde for `Error` and added https://github.com/FuelLabs/fuel-vm/issues/118

**Review by kayagokalp — APPROVED** [review]
LGTM

**Review by adlerjohn — APPROVED** [review]
utACK

**Review by Voxelot — APPROVED** [review]

**Pull request merged**